### PR TITLE
Users/sharathm/azurebatchcf

### PR DESCRIPTION
**Pull request by sharathmalladi**
This pull request includes prior work for running counterfactual evaluations on Azure Batch as well as the Important Features work.

### CODE REVIEW

File: DataScience/ImportantFeatures.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+import os, argparse, sys
+from subprocess import check_output, DEVNULL
+
+def extract_features(fp, inv_hash):
+    features = []
+    text = open(fp).read().split('\n:0\n',1)[1].strip()
+    if '\n' not in text:
+        print ('no features found in model output file: {0}'.format(fp))
+    else:
+        for line in text.split('\n'):
```
Comment (dwaijam) [inline DataScience/ImportantFeatures.py]: splitlines ?

File: DataScience/ImportantFeatures.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+    print('\n=====================================')
+    print('Testing a range of L1 regularization')
+    all_features_funnel = []
+    index = 0
+    max_run_count = 20
+    while True:
+        readModel_fp = readModel_fp_base.format(index)
+        vw_readable_model_cmd_base = vw_base + ' --readable_model {0}'.format(readModel_fp)
+        vw_readable_model_cmd = vw_readable_model_cmd_base + ' -c --l1 {0}'.format(index)
```
Comment (marco-rossi29) [inline DataScience/ImportantFeatures.py]: Shouln't the l1 variable be used somewhere in this line?

File: DataScience/ImportantFeatures.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        if num_features < min_num_features:
+            print('Number of features is {0} which is below the minimum of {1}. Exiting the loop with L1 value of: {2:.0e}'.format(num_features, min_num_features, l1))
+            break
+            
+        # Add a max run count so we avoid getting stuck in an infinite loop for any special case.
+        if index > max_run_count:
+            print('Run count exceeds max run count. Exiting the loop with L1 value of: {0:.0e}'.format(l1))
+            break
+        else:
+            l1 *= 10
```
Comment (marco-rossi29) [inline DataScience/ImportantFeatures.py]: why are you multipling this if l1 is never used for your cmd: `vw_readable_model_cmd`

File: DataScience/ImportantFeatures.py (hunk 2 above)
Comment (sharathmalladi) [reply DataScience/ImportantFeatures.py]: You are right. It is a bug.

---
In reply to: [279973367](https://github.com/Microsoft/mwt-ds/pull/132#discussion_r279973367) [](ancestors = 279973367)

File: DataScience/Experimentation.py [hunk 4]
```diff
@@ -265,19 +271,19 @@ def generate_predictions_files(log_fp, policies):
...
+    print('Learning rates: ['+', '.join(map(str,args.learning_rates))+']')
+    print('L1 regularization: ['+', '.join(map(str,args.regularizations))+']')
+    print('Power_t rates: ['+', '.join(map(str,args.power_t_rates))+']')
     print()
     print("Shared feature namespaces: " + str(shared_features))
     print("Action feature namespaces: " + str(action_features))
     print("Marginal feature namespaces: " + str(marginal_features))
     print("***************************************")
-    if input('Press ENTER to start (any other key to exit)...' ) != '':
+    if __name__ == '__main__' and input('Press ENTER to start (any other key to exit)...' ) != '':
```
[the remark this answers is not in the document]
Comment (sharathmalladi) [reply DataScience/Experimentation.py]: Looking at the following link, this can happen if some one is calling Experimentation.py directly instead of calling Experimentation.main() from somewhere else. That said, I am not experienced in python. If you think there are any other concerns, I am happy to address.

[https://medium.freecodecamp.org/whats-in-a-python-s-name-506262fe61e8](https://medium.freecodecamp.org/whats-in-a-python-s-name-506262fe61e8) 

---
In reply to: [279962265](https://github.com/Microsoft/mwt-ds/pull/132#discussion_r279962265) [](ancestors = 279962265)

File: DataScience/ImportantFeatures.py (hunk 3 above)
Comment (sharathmalladi) [reply DataScience/ImportantFeatures.py]: Fixed it by using l1 instead of index in vw_readable_model_cmd.

File: DataScience/Experimentation.py [hunk 5]
```diff
@@ -190,21 +202,12 @@ def generate_predictions_files(log_fp, policies):
...
-        sys.exit()
-
-    parser = argparse.ArgumentParser()
+def add_parser_args(parser):
     parser.add_argument('-f','--file_path', help="data file path (.json or .json.gz format - each line is a dsjson)", required=True)
     parser.add_argument('-b','--base_command', help="base Vowpal Wabbit command (default: vw --cb_adf --dsjson -c )", default='vw --cb_adf --dsjson -c ')
     parser.add_argument('-p','--n_proc', type=int, help="number of parallel processes to use (default: physical processors)", default=psutil.cpu_count(logical=False))
-    parser.add_argument('-s','--shared_namespaces', type=str, help="shared feature namespaces; e.g., 'abc' means namespaces a, b, and c (default: auto-detect from data file)", default='')
-    parser.add_argument('-a','--action_namespaces', type=str, help="action feature namespaces (default: auto-detect from data file)", default='')
+    parser.add_argument('-sn','--shared_namespaces', type=str, help="shared feature namespaces; e.g., 'abc' means namespaces a, b, and c (default: auto-detect from data file)", default='')
```
[the remark this answers is not in the document]
Comment (sharathmalladi) [reply DataScience/Experimentation.py]: Fixed.

---
In reply to: [279961630](https://github.com/Microsoft/mwt-ds/pull/132#discussion_r279961630) [](ancestors = 279961630)

File: DataScience/Experimentation.py [hunk 6]
```diff
@@ -190,21 +202,12 @@ def generate_predictions_files(log_fp, policies):
...
+    parser.add_argument('-an','--action_namespaces', type=str, help="action feature namespaces (default: auto-detect from data file)", default='')
```
[the remark this answers is not in the document]
Comment (sharathmalladi) [reply DataScience/Experimentation.py]: Fixed.

---
In reply to: [279961702](https://github.com/Microsoft/mwt-ds/pull/132#discussion_r279961702) [](ancestors = 279961702)

File: DataScience/LogDownloader.py [hunk 7]
```diff
@@ -42,51 +42,40 @@ def cmp_files(f1, f2, start_range_f1=0, start_range_f2=0, erase_checkpoint_line=
                 return True
             prev_b1 = b1
 
-def parse_argv(argv):
-
-    parser = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter)
+def add_parser_args(parser):
     parser.add_argument('-a','--app_id', help="app id (i.e., Azure storage blob container name)", required=True)
     parser.add_argument('-l','--log_dir', help="base dir to download data (a subfolder will be created)", required=True)
+    parser.add_argument('-cs','--conn_string', help="storage account connection string", required=True)
```
[the remark this answers is not in the document]
Comment (sharathmalladi) [reply DataScience/LogDownloader.py]: Fixed.

---
In reply to: [279965216](https://github.com/Microsoft/mwt-ds/pull/132#discussion_r279965216) [](ancestors = 279965216)

File: DataScience/LogDownloader.py [hunk 8]
```diff
@@ -42,51 +42,40 @@ def cmp_files(f1, f2, start_range_f1=0, start_range_f2=0, erase_checkpoint_line=
...
     parser.add_argument('-s','--start_date', help="downloading start date (included) - format YYYY-MM-DD", type=valid_date)
     parser.add_argument('-e','--end_date', help="downloading end date (included) - format YYYY-MM-DD", type=valid_date)
     parser.add_argument('-o','--overwrite_mode', type=int, help='''    0: never overwrite; ask the user whether blobs are currently used [default]
     1: ask the user how to proceed when the files have different sizes or when the blobs are currently being used
     2: always overwrite; download currently used blobs
     3: never overwrite; append if the size is larger, without asking; download currently used blobs
     4: never overwrite; append if the size is larger, without asking; skip currently used blobs''', default=0)
-    parser.add_argument('--dry_run', help="print which blobs would have been downloaded, without downloading", action='store_true')
+    parser.add_argument('--dry_run', help="print which blobs would have been downloaded, without downloading", action='store_true', default=False)
```
[the remark this answers is not in the document]
Comment (sharathmalladi) [reply DataScience/LogDownloader.py]: Fixed

---
In reply to: [279965406](https://github.com/Microsoft/mwt-ds/pull/132#discussion_r279965406) [](ancestors = 279965406)

File: DataScience/LogDownloader.py [hunk 9]
```diff
@@ -42,51 +42,40 @@ def cmp_files(f1, f2, start_range_f1=0, start_range_f2=0, erase_checkpoint_line=
...
     parser.add_argument('--create_gzip_mode', type=int, help='''Mode to create gzip file(s) for Vowpal Wabbit:
     0: create one gzip file for each LastConfigurationEditDate prefix
     1: create a unique gzip file by merging over file dates
     2: create a unique gzip file by uniquing over EventId and sorting by Timestamp''', default=-1)
     parser.add_argument('--delta_mod_t', type=int, default=3600, help='time window in sec to detect if a file is currently in use (default=3600 - 1 hour)')
     parser.add_argument('--max_connections', type=int, default=4, help='number of max_connections (default=4)')
-    parser.add_argument('--verbose', help="print more details", action='store_true')
-    parser.add_argument('--confirm', help="confirm before downloading", action='store_true')
+    parser.add_argument('--verbose', help="print more details", action='store_true', default=False)
```
[the remark this answers is not in the document]
Comment (sharathmalladi) [reply DataScience/LogDownloader.py]: Fixed.

---
In reply to: [279965882](https://github.com/Microsoft/mwt-ds/pull/132#discussion_r279965882) [](ancestors = 279965882)

File: DataScience/LogDownloader.py [hunk 10]
```diff
@@ -42,51 +42,40 @@ def cmp_files(f1, f2, start_range_f1=0, start_range_f2=0, erase_checkpoint_line=
...
+    parser.add_argument('--confirm', help="confirm before downloading", action='store_true', default=False)
```
[the remark this answers is not in the document]
Comment (sharathmalladi) [reply DataScience/LogDownloader.py]: Fixed.

---
In reply to: [279965900](https://github.com/Microsoft/mwt-ds/pull/132#discussion_r279965900) [](ancestors = 279965900)

File: DataScience/LogDownloader.py [hunk 11]
```diff
@@ -291,14 +276,14 @@ def download_container(app_id, log_dir, start_date=None, end_date=None, overwrit
                                 for fp in selected_fps_merged:
                                     print('Adding: {}'.format(fp))
                                     with open(fp, 'rb') as f_in:
-                                        shutil.copyfileobj(f_in, f_out, length=100*1024**2)   # writing chunks of 100MB to avoid consuming memory
+                                        shutil.copyfileobj(f_in, f_out, length=1024**3)   # writing chunks of 1GB to avoid consuming memory
```
[the remark this answers is not in the document]
Comment (sharathmalladi) [reply DataScience/LogDownloader.py]: Changing it to 1GB was faster on the azure batch vm where this was running.

---
In reply to: [279966839](https://github.com/Microsoft/mwt-ds/pull/132#discussion_r279966839) [](ancestors = 279966839)

File: DataScience/LogDownloader.py [hunk 12]
```diff
@@ -42,51 +42,40 @@ def cmp_files(f1, f2, start_range_f1=0, start_range_f2=0, erase_checkpoint_line=
...
-    return kwargs
 
 def update_progress(current, total):
     barLength = 50 # Length of the progress bar
     progress = current/total
-    block = int(barLength*progress)
-    text = "\rProgress: [{0}] {1:.1f}%".format( "#"*block + "-"*(barLength-block), progress*100)
-    sys.stdout.write(text)
-    sys.stdout.flush()
+    if int(progress*1000)%10 == 0:
```
[the remark this answers is not in the document]
Comment (sharathmalladi) [reply DataScience/LogDownloader.py]: When a large file is downloaded, the azure batch log file was becoming too large due to the # in the log file. So Dwai did this to reduce that amount of logging.

---
In reply to: [279966114](https://github.com/Microsoft/mwt-ds/pull/132#discussion_r279966114) [](ancestors = 279966114)

File: DataScience/ImportantFeatures.py (hunk 3 above)
Comment (sharathmalladi) [reply DataScience/ImportantFeatures.py]: Fixed.

---
In reply to: [279973548](https://github.com/Microsoft/mwt-ds/pull/132#discussion_r279973548) [](ancestors = 279973548)

File: DataScience/LogDownloader.py [hunk 13]
```diff
@@ -291,14 +276,14 @@ def download_container(app_id, log_dir, start_date=None, end_date=None, overwrit
...
+                                        shutil.copyfileobj(f_in, f_out, length=1024**3)   # writing chunks of 1GB to avoid consuming memory
                 elif create_gzip_mode == 2:
                     selected_fps.sort(key=lambda x : (list(map(int,x.split('_data_')[1].split('_')[:3])), -os.path.getsize(x), x))
                     start_date = '-'.join(selected_fps[0].split('_data_')[1].split('_')[:3])
                     end_date = '-'.join(selected_fps[-1].split('_data_')[1].split('_')[:3])
                     output_fp = os.path.join(log_dir, app_id, app_id+'_deepmerged_data_'+start_date+'_'+end_date+'.json.gz')
                     print('Merge, unique, sort, and zip files of all LastConfigurationEditDate to: {}'.format(output_fp))
-                    if not os.path.isfile(output_fp) or input('Output file already exits. Do you want to overwrite [Y/n]? '.format(output_fp)) in {'Y', 'y'}:
+                    if not os.path.isfile(output_fp) or __name__ == '__main__' and input('Output file already exits. Do you want to overwrite [Y/n]? '.format(output_fp)) in {'Y', 'y'}:
```
[the remark this answers is not in the document]
Comment (sharathmalladi) [reply DataScience/LogDownloader.py]: Replied on a different comment about the use of __name__ == '__main__'.

---
In reply to: [279966892](https://github.com/Microsoft/mwt-ds/pull/132#discussion_r279966892) [](ancestors = 279966892)

File: DataScience/LogDownloader.py [hunk 14]
```diff
@@ -281,7 +266,7 @@ def download_container(app_id, log_dir, start_date=None, end_date=None, overwrit
                     end_date = '-'.join(selected_fps_merged[-1].split('_data_')[1].split('_')[:3])
                     output_fp = os.path.join(log_dir, app_id, app_id+'_merged_data_'+start_date+'_'+end_date+'.json.gz')
                     print('Merge and zip files of all LastConfigurationEditDate to: {}'.format(output_fp))
-                    if not os.path.isfile(output_fp) or input('Output file already exits. Do you want to overwrite [Y/n]? '.format(output_fp)) in {'Y', 'y'}:
+                    if not os.path.isfile(output_fp) or __name__ == '__main__' and input('Output file already exits. Do you want to overwrite [Y/n]? '.format(output_fp)) in {'Y', 'y'}:
```
[the remark this answers is not in the document]
Comment (sharathmalladi) [reply DataScience/LogDownloader.py]: Replied on a different comment.

---
In reply to: [279966621](https://github.com/Microsoft/mwt-ds/pull/132#discussion_r279966621) [](ancestors = 279966621)

File: DataScience/AzureUtil.py [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+            print('Upload Time:',(t2-t1)-timedelta(microseconds=(t2-t1).microseconds))
+        except Exception as e:
+            print(e)
+            if throw_ex: raise(e)
+            
+    def download_from_blob(self, storage_container_name, storage_file_name, local_file_path, throw_ex = False):
+        try:
+            print("\nDownloading from Blob storage to file")
+            t1 = datetime.now()
+            self.block_blob_service.get_blob_to_path(storage_container_name, storage_file_name, local_file_path)
```
[the remark this answers is not in the document]
Comment (sharathmalladi) [reply DataScience/AzureUtil.py]: I looked at the option refactor LogDownloader to call into AzureUtil methods. The ones in AzureUtil dont give the same level of fine-grained control when downloading the blobs (like overwriting the existing files). So then the question might be why not just call LogDownloader. The LogDownloader class has lot of code that is specific to logs (dependent on the blob name, path and properties for example). Also, LogDownloader has a lot of logic for each downloaded file and AzureUtil operations are more simple. I think the effort needed to refactor LogDownloader is quite a lot that I feel it is ok to leave it as is.

File: DataScience/AzureUtil.py [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+            print(storage_file_name)
+            print("Done downloading blob")
+            print('Download Time:',(t2-t1)-timedelta(microseconds=(t2-t1).microseconds))
+        except Exception as e:
+            print(e)
+            if throw_ex: raise(e)
+            
+    def download_all_blobs(self, storage_container_name, local_dir, throw_ex = False):
+        generator = self.block_blob_service.list_blobs(storage_container_name)
+        for blob in generator:
```
[the remark this answers is not in the document]
Comment (sharathmalladi) [reply DataScience/AzureUtil.py]: Replied in detail on a different comment.

File: DataScience/LogDownloader.py [hunk 17]
```diff
@@ -42,51 +42,40 @@ def cmp_files(f1, f2, start_range_f1=0, start_range_f2=0, erase_checkpoint_line=
...
     parser.add_argument('--delta_mod_t', type=int, default=3600, help='time window in sec to detect if a file is currently in use (default=3600 - 1 hour)')
     parser.add_argument('--max_connections', type=int, default=4, help='number of max_connections (default=4)')
-    parser.add_argument('--verbose', help="print more details", action='store_true')
-    parser.add_argument('--confirm', help="confirm before downloading", action='store_true')
+    parser.add_argument('--verbose', help="print more details", action='store_true', default=False)
+    parser.add_argument('--confirm', help="confirm before downloading", action='store_true', default=False)
     parser.add_argument('-v','--version', type=int, default=2, help='''version of log downloader to use:
     1: for uncooked logs (only for backward compatibility) [deprecated]
     2: for cooked logs [default]''')
-
```
[the remark this answers is not in the document]
Comment (sharathmalladi) [reply DataScience/LogDownloader.py]: I think this comment was regarding removing the use of ConfigParser and making the connectionstring mandatory. I made it optional now and added back the config parser as the default behavior. Could you run this option as I don't have a ds.config file to validate this code path.

File: DataScience/LogDownloader.py [hunk 18]
```diff
@@ -257,7 +242,7 @@ def download_container(app_id, log_dir, start_date=None, end_date=None, overwrit
                         end_date = '-'.join(models[model][-1].split('_data_')[1].split('_')[:3])
                         output_fp = os.path.join(log_dir, app_id, app_id+'_'+model+'_data_'+start_date+'_'+end_date+'.json.gz')
                         print('Concat and zip files of LastConfigurationEditDate={} to: {}'.format(model, output_fp))
-                        if os.path.isfile(output_fp) and input('Output file already exits. Do you want to overwrite [Y/n]? '.format(output_fp)) not in {'Y', 'y'}:
+                        if os.path.isfile(output_fp) and __name__ == '__main__' and input('Output file already exits. Do you want to overwrite [Y/n]? '.format(output_fp)) not in {'Y', 'y'}:
```
[the remark this answers is not in the document]
Comment (sharathmalladi) [reply DataScience/LogDownloader.py]: Replied on a different comment in detail about this.

File: DataScience/Experimentation.py [hunk 19]
```diff
@@ -206,8 +206,8 @@ def add_parser_args(parser):
...
     parser.add_argument('-b','--base_command', help="base Vowpal Wabbit command (default: vw --cb_adf --dsjson -c )", default='vw --cb_adf --dsjson -c ')
     parser.add_argument('-p','--n_proc', type=int, help="number of parallel processes to use (default: physical processors)", default=psutil.cpu_count(logical=False))
-    parser.add_argument('-sn','--shared_namespaces', type=str, help="shared feature namespaces; e.g., 'abc' means namespaces a, b, and c (default: auto-detect from data file)", default='')
-    parser.add_argument('-an','--action_namespaces', type=str, help="action feature namespaces (default: auto-detect from data file)", default='')
+    parser.add_argument('--shared_namespaces', type=str, help="shared feature namespaces; e.g., 'abc' means namespaces a, b, and c (default: auto-detect from data file)", default='')
+    parser.add_argument('--action_namespaces', type=str, help="action feature namespaces (default: auto-detect from data file)", default='')
     parser.add_argument('-m','--marginal_namespaces', type=str, help="marginal feature namespaces (default: auto-detect from data file)", default='')
```
Comment (marco-rossi29) [inline DataScience/Experimentation.py]: Let's remove also this '-m' for consistency

File: DataScience/ExperimentationAzure.py [hunk 20]
```diff
@@ -137,7 +137,7 @@ def check_system():
         unknown.append('--model')
         unknown.append(model_fp)
         unknown.append('--min_num_features')
-        unknown.append('3')
+        unknown.append('1')
```
Comment (marco-rossi29) [inline DataScience/ExperimentationAzure.py]: why this change?

File: DataScience/ImportantFeatures.py [hunk 21]
```diff
@@ -113,7 +112,7 @@ def get_important_features(log_file, ml_args, warmstart_model=None, min_num_feat
     while True:
         readModel_fp = readModel_fp_base.format(index)
```
Comment (marco-rossi29) [inline DataScience/ImportantFeatures.py]: I would use directly `log_file+'.readModel.{0}.txt'` here for readibility removing the variable `readModel_fp_base`

File: DataScience/ExperimentationAzure.py (hunk 20 above)
Comment (sharathmalladi) [reply DataScience/ExperimentationAzure.py]: I added the min_num_features initially to ensure I don't get back zero buckets. Tyler's point was that setting it to 3 will mean we cannot have a bucket of just 1 - which can happen (depending on the data). So I changed it to 1.

File: DataScience/Experimentation.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (marco-rossi29) [reply DataScience/Experimentation.py]: If `__name__` is inside a `def` it will never be `__main__` no matter how you call the parent script

[the remark this answers is not in the document]
Comment (marco-rossi29) [reply DataScience/Experimentation.py]: You're right. Your usage is the right one!

**Comment by marco-rossi29** [comment]
@sharathmalladi Thank you for all the changes.

I'm going to run some tests on my side to ensure LogDownloader and Experimentation are working correctly and then merge.

In the meantime, I'll think if there is a simple way to avoid the code duplication between `LogDownloader` and `AzureUtil`

File: DataScience/LogDownloader.py (hunk 17 above)
[the remark this answers is not in the document]
Comment (marco-rossi29) [reply DataScience/LogDownloader.py]: No, this comment is about behavior of kwargs when `'version' == 1`

File: DataScience/LogDownloader.py (hunk 12 above)
[the remark this answers is not in the document]
Comment (marco-rossi29) [reply DataScience/LogDownloader.py]: The progress bar is now not fluid. It updates only 3 times. Please fix

My suggestion is not to call `update_progress` if `'__name__' != '__main__'`

File: DataScience/ExperimentationAzure.py (hunk 20 above)
Comment (marco-rossi29) [reply DataScience/ExperimentationAzure.py]: Can we rename the variable `unknown` to have a less unknown meaning...

Comment (sharathmalladi) [reply DataScience/ExperimentationAzure.py]: Changed to other_args

File: DataScience/LogDownloader.py [hunk 22]
```diff
@@ -69,7 +69,7 @@ def add_parser_args(parser):
 def update_progress(current, total):
     barLength = 50 # Length of the progress bar
     progress = current/total
-    if int(progress*1000)%10 == 0: 
+    if int(progress*1000)%10 == 0 or '__name__' == '__main__':
```
Comment (marco-rossi29) [inline DataScience/LogDownloader.py]: This shouldn't be the place where we address if writing or not to. update_progress() shouldn't be called if no display is needed

Comment (sharathmalladi) [reply DataScience/LogDownloader.py]: Added a default behavior to report progress for LogDownloader

**Comment by sharathmalladi** [comment]
> The more I think about it, the more I feel we can actually get away without introducing the `report_progress` input variable and just use:
> `process_checker = update_progress if '__name__' == '__main__' else None`
> 
> What do you think?

Since LogDownloader can be called from other scripts (usually is), I did not want to add dependency on __main__ since then it will only work if you call LogDownloader directly from python. I wanted to leave it to the caller to decide whether or not to report progress. So I added this report_progress flag - it defaults to true (store_false) if not specified. That way for all other existing scripts, the behavior is not impacted (like you wanted). However for my purposes, I set this flag to false when called from ExperimentationAzure since it is running in Azure batch in a non-interactive fashion.

**Comment by sharathmalladi** [comment]
> ouln't the default w



> Shouln't the default we `False`?

The default is set to true (since you did not want the experience to change for other callers - and this script was essentially logging every time prior to my changes).

**Comment by sharathmalladi** [comment]
> I actually think that there is a small bug, since you never use `args.report_progess`
> We should remove this line, and it will be goodvin my opinion

I do use report_progress. download_container() accepts a parameter for report_progress which is being set inside ExperimentationAzure.py script as argument input to this LogDownloader.

File: DataScience/ExperimentationAzure.py [hunk 23]
```diff
@@ -3,7 +3,7 @@
 from subprocess import check_output, STDOUT
 from AzureUtil import AzureUtil
 import Experimentation
-import ImportantFeatures
+import FeatureImportance
```
Comment (marco-rossi29) [inline DataScience/ExperimentationAzure.py]: It should be FeaturesImportance since there is usually more than 1 feature

File: DataScience/ExperimentationAzure.py [hunk 24]
```diff
@@ -36,8 +36,8 @@ def check_system():
     main_parser.add_argument('--run_experimentation', help="run Experimentation.py", action='store_true')
     main_parser.add_argument('--delete_logs_dir', help="delete logs directory before starting to download new logs", action='store_true')
     main_parser.add_argument('--cleanup', help="delete logs and created files after use", action='store_true')
-    main_parser.add_argument('--get_important_features', help="run ImportantFeatures.py", action='store_true')
-    main_parser.add_argument('--importantfeatures_filename', help="name of the output feature importance file", default='featureimportance.json')
+    main_parser.add_argument('--get_feature_importance', help="run FeatureImportance.py", action='store_true')
+    main_parser.add_argument('--feature_importance_filename', help="name of the output feature importance file", default='featureimportance.json')
     main_parser.add_argument('--ml_args', help="the online policy that we need for calculating the feature importances", required=True)
```
Comment (marco-rossi29) [inline DataScience/ExperimentationAzure.py]: `feature importances` -> `features importance`

File: DataScience/ExperimentationAzure.py (hunk 23 above)
Comment (sharathmalladi) [reply DataScience/ExperimentationAzure.py]: Wont fix after discussing with Marco.

---
In reply to: [283940769](https://github.com/microsoft/mwt-ds/pull/132#discussion_r283940769) [](ancestors = 283940769)
